content/post/078-go-ipfs-0.4.23.md

```diff
@@ -0,0 +1,84 @@
+---
+date: 2020-01-27
+url: 2020-01-27-go-ipfs-0-4-23
+title: go-ipfs 0.4.23 released
+author: Alan Shaw
+---
+
+> Would sir/madam care for another patch release while they wait?
+
+Yes that's right, the next feature release of go-ipfs (0.5.0) is, well, running a tiny bit behind schedule. In the mean time though we have patches, and I'm not talking pirate eye patches, I'm talking bug fixes. We're hunting these bugs like they're Pokemon, and jeez, do we come across some rare and difficult to fix ones? - you betcha.
+
+Alright, enough funny business, what's the deal? Ok so, I don't want to _alarm_ anyone but this release has some **critical fixes** and if you're using go-ipfs or know someone who is then you and your friends need to slide into your upgrade pants and give those IPFS nodes a good wipe down ASAP.
+
+## 🔦 Highlights
+
+### 🤝 Fixed Spontaneous TLS Disconnects
+
+If this isn't reason enough to upgrade I don't know what is. Turns out, a TLS handshake _may_ have accidentially been unintentionally aborted for no good reason 😱. I shouldn't need to tell you this but \*whispers\* that's really bad for connectivity! Excellent news though, it's fixed, and if you want to stay well connected in the glorious future of p2p communications then you should definitely upgrade.
+
+### 😱 Fixed Panics and Crashes
+
+Panicing won't help, in life, and also in golang. Stay calm and breathe slowly. We patched a number of panics and crashes that were uncovered, including a panic seen commonly in the websocket transport. High ten 🙌?
+
+### 🔁 Fixed Resursive Resolving of dnsaddr Multiaddrs
+
+Imagine this: rolling out a brand spanking new set of bootstrap nodes only to discover their new addresses are not resolvable. I know right - bad news bears 🐻!?
+
+Ok, "bear" this in mind: you want to keep all your DNS TXT records [below 512 bytes to avoid UDP fragmentation](https://serverfault.com/questions/840241/do-dns-queries-always-travel-over-udp), otherwise you'll get a truncated reply and have to connect with TCP to get all the records. If you have lots of dnsaddr TXT records then it can be more efficient to use recursive resolving than to get a truncated reply and go through the famous 18-way SYN, SYN-ACK ACK, ACK-SYN, ACK-ACK (...etc, etc) TCP handshake, not to mention the fact that go-ipfs will not even try to fallback to TCP 😅.
+
+Anyway, long story short. We fixed that. You're welcome.
+
+### 📻 Retuned Connection Manager
+
+The Connection Manager has been tuned to better prioritise existing connections by not counting new connections in the "grace" period towards connection limits. New connections are like new friends. You can't hang out with everyone all the time, I mean, it just gets difficult to book a resturant after a while.
```

Comment:
We should try to explain the problem with the current system. Specifically, nodes receiving too many inbound connections would have more connections in the grace period (30s) than the connection limit. When this happens, we would disconnect form _all_ peers not in the grace period, regardless of how useful they were.